app/controllers/products_controller.rb

```diff
@@ -1,62 +1,76 @@
 class ProductsController < ApplicationController
-  before_action :find_product, only: [:show, :edit, :update, :destroy]
+  before_action :find_product, only: [:show, :edit, :update, :destroy, :toggle_active, :toggle_inactive]
 
   def index
-    # if params[:merchant_id]
-    #   merchant = Merchant.find_by(id: params[:merchant_id])
-    #   @products = merchant.products
-    # elsif params[:category_id]
-    #   category = Category.find_by(id: params[:category_id])
-    #   @products = category.products
-    # else
-    #   @products = Product.all
-    # end
-    @products = Product.where(status: true)
+    if params[:merchant_id]
+      merchant = Merchant.find_by(id: params[:merchant_id])
+      if merchant == nil
+        return head :not_found
+      else
+        # merchant = Merchant.find_by(id: params[:merchant_id])
+        @products = merchant.products
+      end
+    elsif params[:category_id]
+      category = Category.find_by(id: params[:category_id])
+      if category == nil
+        return head :not_found
+      else
+        @products = category.products
+      end
+    else
+      @products = Product.where(status: true)
+    end
+    # @products = Product.where(status: true)
   end
 
   def new
     @product = Product.new
   end
```

Comment:
I updated this action to take into consideration of the nested routes for merchant and product.